src/functions.php

```diff
@@ -31,6 +46,14 @@ function reject($promiseOrValue = null)
         });
     }
 
+    if ($promiseOrValue instanceof AsyncInteropPromise) {
+        return new Promise(function ($resolve, $reject) use ($promiseOrValue) {
+            $promiseOrValue->when(function ($reason = null, $value = null) use ($resolve, $reject) {
+                $reject($reason ? $reason : $value);
```

Comment:
> I just wasn't sure because of UnhandledRejectionException::nullOrResolve.

`nullOrResolve` was actually wrong. In case of rejection without a reason, it would call the `when()` callbacks with `null` as first argument signalling success instead of failure. This is fixed in 8073b04.

> RejectedPromise seems to miss a check for AsyncInteropPromise in __construct btw.

Good catch, fixed in 8c68ff6